src/ssh_client.py

```diff
@@ -0,0 +1,77 @@
+import paramiko
+import logging
+
+from src.base_client import BaseShellClient
+
+
+class SSHClient(BaseShellClient):
+    def __enter__(self):
+        try:
+            self.client = paramiko.SSHClient()
+            self.client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+            self.client.connect(self.host, port=self.port, username=self.username, password=self.password)
+            logging.info(f"Connected to {self.host} successfully.")
+        except Exception as e:
+            logging.error(f"Failed to connect to {self.host}: {e}")
+            self.client = None
+        return self
+
+    def upload_file(self, local_path, remote_path):
+        if self.client is None:
+            raise ConnectionError("SSH connection not established.")
+        try:
+            sftp = self.client.open_sftp()
+            sftp.put(local_path, remote_path)
+            sftp.close()
+            logging.info(f"Uploaded {local_path} to {remote_path} successfully.")
+        except Exception as e:
+            logging.error(f"Failed to upload file: {e}")
+
+    def execute_command(self, command):
+        if self.client is None:
+            raise ConnectionError("SSH connection not established.")
+        try:
+            stdin, stdout, stderr = self.client.exec_command(command)
+            return stdout.read().decode('utf-8'), stderr.read().decode('utf-8')
+        except Exception as e:
+            logging.error(f"Failed to execute command: {e}")
+            return "", str(e)
+
+    def execute_script(self, local_script, remote_script, directory):
```

Comment:
Method name does not match what happens inside method. Why did you use implicit steps ( uploading, comparison )  inside current method?
Please split on different functions: upload_file, compare two values ( checksums of original and uploaded file), execute_script